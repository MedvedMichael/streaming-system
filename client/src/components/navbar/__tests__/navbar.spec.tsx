import { render, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ThemeProvider } from 'styled-components'
import { DarkThemeContext } from '../../Providers'
import { lightTheme } from '../../themes'
import Navbar from '../navbar'

jest.mock('next/router', () => ({
    ...jest.requireActual('next/router'),
    default: {
        push: () => null
    },
    useRouter: () => ({
        pathname: ''
    })
}))

test ('should render', () => {
    const res = shallow(
        <DarkThemeContext.Provider value={false}>
            <ThemeProvider theme={lightTheme}>
                <Navbar />
            </ThemeProvider>
        </DarkThemeContext.Provider>)
    expect(toJson(res)).toMatchSnapshot()
    
})