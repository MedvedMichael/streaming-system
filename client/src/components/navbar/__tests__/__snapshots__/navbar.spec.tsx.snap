// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<Component
  theme={
    Object {
      "bg": Object {
        "aboutMe": "linear-gradient(to right, #5e9be0, #c8d8e3)",
        "card": "linear-gradient(to right, #5e9be0, #6fa5e3)",
        "input": "rgba(65,67,78,0.12)",
        "inset": "#5e9be0",
        "nav": "#437dbf",
        "primary": "#86c1f0",
        "secondary": "#5e9be0",
      },
      "border": Object {
        "card": "1px solid #18304d",
      },
      "dark": false,
      "fontSizes": Array [
        "14px",
        "16px",
        "18px",
        "22px",
        "26px",
        "32px",
        "40px",
      ],
      "fontWeights": Object {
        "body": 400,
        "bold": 700,
        "heading": 800,
        "link": 600,
        "subheading": 500,
      },
      "lineHeights": Object {
        "body": 1.5,
        "code": 1.6,
        "heading": 1.3,
      },
      "text": Object {
        "onPrimary": "#ffffff",
        "placeholder": "rgba(82,85,96,0.5)",
        "primary": "#fbfbfc",
        "quarternary": "#9194a1",
        "secondary": "#2f3037",
        "tertiary": "#525560",
      },
      "transition": Object {
        "bg": "background 300ms linear 0s",
        "opacity": "opacity 200ms ease",
        "primary": "all 300ms linear 0s",
      },
    }
  }
>
  <Navbar />
</Component>
`;
